lib/Parse/ParseStmt.cpp

```diff
@@ -1833,6 +1833,74 @@ Parser::parseStmtConditionElement(SmallVectorImpl<StmtConditionElement> &result,
   return Status;
 }
 
+/// Returns `true` if the current token represents the start of a conditional
+/// statement body.
+bool Parser::isStartOfConditionalStmtBody() {
+  Parser::BacktrackingScope Backtrack(*this);
+
+  if (!Tok.is(tok::l_brace)) {
+    // Statement bodies always start with a '{'. If there is no '{', we can't be
+    // at the statement body.
+    return false;
+  }
+
+  skipSingle();
```

Comment:
Nit: At first glance, this `skipSingle();` seems to not be tied to any logic; however, I see now that we want to skip over the brace and enter the condition body. Maybe write it like this to make it more clear:
```cpp
if (Tok.is(tok::l_brace)) {
  skipSingle();
} else {
  // Statement bodies always start with a '{'. If there is no '{', we can't be
  // at the statement body.
  return false;
}
```

Reply:
This sort of stylistic aspects normally goes down to project convention or personal preference, so we often leave this to the author, for example this code as it is makes more sense to me personally than suggested style :)

Reply:
That makes sense, thanks for explaining!